### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by RezenkovD**

### CODE REVIEW

File: tests/factories/categories.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from tests.conftest import session
+
+
+class CategoryFactory(factory.alchemy.SQLAlchemyModelFactory):
+    class Meta:
+        model = Category
+        sqlalchemy_session = session
+        sqlalchemy_session_persistence = "commit"
+
+    id = factory.Faker("id")
```
Comment (qqlexa) [inline tests/factories/categories.py:13]: You may not create the `id` field in Factories because DB does it.

File: tests/factories/items.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from tests.conftest import session
+
+
+class ItemFactory(factory.alchemy.SQLAlchemyModelFactory):
+    class Meta:
+        model = Item
+        sqlalchemy_session = session
+        sqlalchemy_session_persistence = "commit"
+
+    id = factory.Faker("id")
```
Comment (qqlexa) [inline tests/factories/items.py:13]: The same

File: tests/factories/users.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+from tests.conftest import session
+
+
+class UserFactory(factory.alchemy.SQLAlchemyModelFactory):
+    class Meta:
+        model = User
+        sqlalchemy_session = session
+        sqlalchemy_session_persistence = "commit"
+
+    id = factory.Faker("id")
```
Comment (qqlexa) [inline tests/factories/users.py:13]: The same

File: tests/test_endpoints/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+from .test_categories import test_create_category_for_user
+
+# from .test_items import test_create_item_for_user
```
Comment (qqlexa) [inline tests/test_endpoints/__init__.py:3]: Is it useful?

File: tests/test_endpoints/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+# from .test_users import test_create_user
```
Comment (qqlexa) [inline tests/test_endpoints/__init__.py:4]: Is it useful?

File: tests/test_endpoints/test_items.py [hunk 6]
```diff
@@ -1,10 +1,14 @@
...
+from tests.factories.users import UserFactory
+from tests.factories.categories import CategoryFactory
 
 
 def test_create_item_for_user(db):
-    from tests.test_categories import test_create_category_for_user
+    user = UserFactory.create(id=1, email="test@gmail.com", hashed_password="test89A")
 
-    test_create_category_for_user(db)
+    category_one = CategoryFactory.create(id=1, user_id=1, title="Accessories")
```
Comment (qqlexa) [inline tests/test_endpoints/test_items.py:10]: Don't give `id` to CategoryFactory or any other. DB knows id that you need.
```python
category_one = CategoryFactory.create(user_id=user.id, title="Accessories")
```

File: tests/test_endpoints/test_items.py [hunk 7]
```diff
@@ -1,10 +1,14 @@
-from tests.conftest import client, db
+from tests.conftest import client
+
+from tests.factories.users import UserFactory
+from tests.factories.categories import CategoryFactory
 
 
 def test_create_item_for_user(db):
-    from tests.test_categories import test_create_category_for_user
+    user = UserFactory.create(id=1, email="test@gmail.com", hashed_password="test89A")
```
Comment (qqlexa) [inline tests/test_endpoints/test_items.py:8]: Let's use default Faker values
```python
user = UserFactory.create()  # it means that id will be Faker(int) and etc.
```

File: tests/test_endpoints/test_statistics.py [hunk 8]
```diff
@@ -35,9 +70,40 @@ def test_get_stats(db):
 
 
 def test_get_stats_month(db):
-    from tests.test_items import test_create_item_for_user
+    user_one = UserFactory.create(
+        id=1, email="test@gmail.com", hashed_password="test89A"
+    )
+    user_two = UserFactory.create(
+        id=2, email="test2@gmail.com", hashed_password="test89B"
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:77]: You should not control id manually

File: tests/test_endpoints/test_statistics.py [hunk 9]
```diff
@@ -35,9 +70,40 @@ def test_get_stats(db):
 
 
 def test_get_stats_month(db):
-    from tests.test_items import test_create_item_for_user
+    user_one = UserFactory.create(
+        id=1, email="test@gmail.com", hashed_password="test89A"
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:74]: You should not control id manually

File: tests/test_endpoints/test_statistics.py [hunk 10]
```diff
@@ -35,9 +70,40 @@ def test_get_stats(db):
...
+    user_two = UserFactory.create(
+        id=2, email="test2@gmail.com", hashed_password="test89B"
+    )
+
+    category_one = CategoryFactory.create(id=1, user_id=1, title="Accessories")
+    category_two = CategoryFactory.create(id=2, user_id=1, title="Food")
 
-    test_create_item_for_user(db)
+    item_one = ItemFactory.create(
+        id=1,
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:84]: You should not control id manually

File: tests/test_endpoints/test_statistics.py [hunk 11]
```diff
@@ -35,9 +70,40 @@ def test_get_stats(db):
...
+        user_id=1,
+        description="Redmi Buds 3",
+        price=1299,
+        time="2022-11-01T08:52:53.301000",
+        category_id=1,
+    )
+    item_two = ItemFactory.create(
+        id=2,
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:92]: You should not control id manually

File: tests/test_endpoints/test_statistics.py [hunk 12]
```diff
@@ -35,9 +70,40 @@ def test_get_stats(db):
...
+        user_id=1,
+        description="Pasta",
+        price=56,
+        time="2022-11-17T08:52:53.301000",
+        category_id=2,
+    )
+    item_three = ItemFactory.create(
+        id=3,
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:100]: You should not control id manually

**Review by qqlexa — CHANGES_REQUESTED** [review]

File: tests/test_endpoints/test_statistics.py [hunk 13]
```diff
@@ -6,38 +6,35 @@
 
 def test_get_stats(db):
     user_one = UserFactory.create(
-        id=1, email="test@gmail.com", hashed_password="test89A"
+        email="test@gmail.com", hashed_password="test89A"
     )
     user_two = UserFactory.create(
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:11]: Do we use it?

File: tests/test_endpoints/test_statistics.py [hunk 14]
```diff
@@ -6,38 +6,35 @@
 
 def test_get_stats(db):
     user_one = UserFactory.create(
-        id=1, email="test@gmail.com", hashed_password="test89A"
+        email="test@gmail.com", hashed_password="test89A"
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:9]: Let's find Faker for email and hashed_password, don't set it manually.

File: tests/test_endpoints/test_users.py [hunk 15]
```diff
@@ -26,34 +26,35 @@ def test_create_user(db):
 
 def test_read_user(db):
     user_one = UserFactory.create(
-        id=1, email="test@gmail.com", hashed_password="test89A"
+        email="test@gmail.com", hashed_password="test89A"
+    )
+    user_two = UserFactory.create(
```
Comment (qqlexa) [inline tests/test_endpoints/test_users.py:31]: Do we need it?

**Review by qqlexa — CHANGES_REQUESTED** [review]

File: tests/test_endpoints/test_statistics.py (hunk 14 above)
Comment (RezenkovD) [reply tests/test_endpoints/test_statistics.py]: For the password - yes. For mail - no. The factory is connected with tests

Comment (qqlexa) [reply tests/test_endpoints/test_statistics.py]: We should not "connect" them. Use variables. user_one.email

File: tests/factories/users.py [hunk 16]
```diff
@@ -11,4 +11,4 @@ class Meta:
         sqlalchemy_session_persistence = "commit"
 
     email = factory.Faker("email")
-    hashed_password = factory.Faker("hashed_password")
+    hashed_password = "hashed_password"
```
Comment (qqlexa) [inline tests/factories/users.py:14]: Use something like that
```python
hashed_password = factory.LazyAttribute(
        lambda o: encrypt(factory.Faker("pystr"))
)
```

**Review by qqlexa — CHANGES_REQUESTED** [review]

File: tests/test_endpoints/test_statistics.py [hunk 17]
```diff
@@ -5,9 +5,10 @@
 
 
 def test_get_stats(db):
-    user_one = UserFactory.create(email="test@gmail.com")
-    user_two = UserFactory.create(email="test2@gmail.com")
-
+    user_one = UserFactory.create()
+    user_two = UserFactory.create()
+    user_one_email = user_one.email
+    user_two_email = user_two.email
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:11]: Just use a `user_two.email`, without declaring a new variable

File: tests/test_endpoints/test_statistics.py [hunk 18]
```diff
@@ -63,9 +64,10 @@ def test_get_stats(db):
 
 
 def test_get_stats_month(db):
-    user_one = UserFactory.create(email="test@gmail.com")
-    user_two = UserFactory.create(email="test2@gmail.com")
-
+    user_one = UserFactory.create()
```
Comment (qqlexa) [inline tests/test_endpoints/test_statistics.py:67]: Better!

**Review by qqlexa — APPROVED** [review]

**Pull request merged**